### ISSUE TITLE
Pinned Shortcut Doubled at Runtime

### ISSUE THREAD

**Issue by JKSnd** [Squirrel/Squirrel.Windows#1094]
Squirrel Version: 1.7.7

If I pin the application immediately after installing it, that pin will work fine. But once an update occurs the pin will launch the application correctly but the pin will appear not to be used. Like so:

![image](https://user-images.githubusercontent.com/17773454/28078279-8cc59732-6621-11e7-8d3c-644a2d05dbca.png)

I assume the pinned shortcut is for the application stub and the running icon is for the executable inside of whichever app folder is relevant. Is it possible to consolidate these?

**Comment by lukeskinner** [comment]
I remember looking at this a while back.. I think there was some code in Squirrel that wasn't setting the appUserModelId when it was rewriting the shortcuts at some point.

The appUserModelId is what allows Windows to know what programs should be grouped on the taskbar. If its not set right on the shortcut or your application then this ends up happening. 

I'll try find some time to see if the issue I found before is still there.

**Comment by msneijders** [comment]
Any known workaround for this problem?

**Comment by lukeskinner** [comment]
The code I mentioned before is at https://github.com/Squirrel/Squirrel.Windows/blob/master/src/Squirrel/UpdateManager.ApplyReleases.cs#L498 

After line 498 it is missing a call to ```shortcut.SetAppUserModelId```. The "com.squirrel.x.x" value it needs will likely need to be created in a variable in the ```ApplyReleases``` method and then passed through to various methods to end up in the ```updateLink``` method.

The only workaround for this would be to take the above squirrel code which updates shortcuts into your app via the onAppUpdate event and then add in the ```shortcut.SetAppUserModelId``` call. It's not much of a workaround - I've put some time aside to try make the PR later today.

**Comment by msneijders** [comment]
Thank you, all taskbar users will appreciate it.

**Comment by lukeskinner** [comment]
I've tested this on version 1.7.7 multiple times and I can't replicate the issue anymore. The above changes I mentioned were needed at some point pre 1.5.0 - looks to have been fixed in the mean time. 

Are you setting the AppUserModelID on startup for your app? 

You should have something like the code below at the earliest possible point in your code (startup/init). 

The PackageName in the string below should be the same as your nuget package name used for squirrel. The ExeName in the string below should be the running applications exe name without the ending .exe and without spaces. 

In the sample app I made to test this it would be "com.squirrel.SampleApp.SampleApp" as both the nuget package and exe name are SampleApp. 

Example code:
```c#
[DllImport("shell32.dll", SetLastError = true)]
static extern void SetCurrentProcessExplicitAppUserModelID([MarshalAs(UnmanagedType.LPWStr)] string AppID);

SetCurrentProcessExplicitAppUserModelID("com.squirrel.PackageName.ExeName");

```

**Comment by JKSnd** [comment]
Ahh, we are not using an explicit AppUserModelID. I'll introduce this and see what happens.

Is this requirement captured in the documentation and I just missed it, or has it been omitted?

**Comment by lukeskinner** [comment]
The AppUserModelID is mentioned in other repos such as Electron in the issue tracker. I haven't seen it anywhere in this repo. If it does fix it then the squirrel.windows documentation can be updated to explain it better. 

https://github.com/electron/electron/issues/4381

**Comment by JKSnd** [comment]
Yep, this did resolve the behaviour. Something of note, however is that the pin had to be re-made in order for the change to be effective. The pin that was made using a version of the app that didn't have its AppUserModelID set wasn't compatible. After upgrading to a version with an explicit AUMID I removed the old pin and made a new one. Then I caused a version bump and a release and tried again. All good from that point forward.

Thanks for your help!
